tmt/steps/report/reportportal.py

```diff
@@ -449,13 +461,24 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
 
         self.check_options()
 
-        launch_time = self.time()
+        # If SSL verification is disabled, do not print warnings with urllib3
+        if not self.data.ssl_verify:
+            urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+            self.warn("SSL verification is disabled for all requests being made to ReportPortal "
+                      f"instance ({self.data.url}).")
```

Comment:
Not just for ReportPortal: warnings will remain disabled for all code from this point forward, i.e. all `report` plugins invoked after this instance, and all `finish` and cleanup code of tmt itself. Can it be undone when this plugin finishes its work? The safest would be wrapping it all in a `try`/`finally` block and undo the act before leaving the function, but it's already so indented and so spaghettified that this way may lead to an extremely ugly diff...

Reply:
Thanks for the hint. You are right we should re-enable the urllib2 warnings.

I agree the diff would be huge due to another indent.

What about using a e.g. context manager? We could move most of the code from the `go()` method into the new method `f(..)` and optionally disable the warnings, something like:

```python
@contextlib.contextmanager
def supress_ssl_warnings(silent=True):
    if silent:
        try:
            # Disable SSL warnings here
            yield
        finally:
            # Enable SSL warnings
            ...
    else:
        yield

def go(...):
    ...
    with supress_ssl_warnings(silent=not self.data.ssl_verify):
        self.f(...)
```

The diff would be still big, but this should not require another indent.

Reply:
Something like this https://github.com/teemtee/tmt/pull/3356/commits/83e4e7889af01ee0712b1e344ffb6e02be8fa2b4 ? What do you think?

Reply:
> Thanks for the hint. You are right we should re-enable the urllib2 warnings.
> 
> I agree the diff would be huge due to another indent.
> 
> What about using a e.g. context manager? We could move most of the code from the `go()` method into the new method `f(..)` and optionally disable the warnings

A context manager would be my choice as well, and yes, splitting the method into two should help manage the indentation. It's an ugly workaround, but the method itself deserves a cleanup, otherwise *everything* will result in an ugly indentation mess :/

Reply:
Yes, I agree.

If you think the diff in this PR is becoming too large, I can separate the patches for `--no-ssl-verify/--ssl-verify` functionality into a separate PR (so we can merge the bug fixes more quickly). Otherwise, I would keep them here.

Thanks!